certification/internal/engine/engine.go

```diff
@@ -302,9 +300,12 @@ func generateBundleHash(bundlePath string) (string, error) {
 		hashBuffer.WriteString(fmt.Sprintf("%s  %s\n", k, files[k]))
 	}
 
-	_, err := artifacts.WriteFile("hashes.txt", &hashBuffer)
-	if err != nil {
-		return "", fmt.Errorf("could not write hash file to artifacts dir: %w", err)
+	artifactsWriter := artifacts.WriterFromContext(ctx)
+	if artifactsWriter != nil {
+		_, err := artifactsWriter.WriteFile("hashes.txt", &hashBuffer)
+		if err != nil {
+			return "", fmt.Errorf("could not write hash file to artifacts dir: %w", err)
+		}
 	}
```

Comment:
[Note To Reviewers]: Here's an example of the new way to write artifacts. This function did not have a context, so it needed to be added in order to implement the new approach.

Reply:
Do we need to worry about `artifactsWriter` being nil in any case? I think the intent here is for when callers do not care about what we write to disk, but what about the cases where we are trying to write to disk, would an exception happen before we even get there?

Reply:
Yep, you're spot on. The `preflight` CLI execution _needs_ the writer to be there, so it throws an error if one is not found before we ever hit this logic. See --> https://github.com/redhat-openshift-ecosystem/openshift-preflight/blob/51df40fe113c71109b1ec967462a1886872d5775/internal/cli/cli.go#L36-L40

For library callers, in theory, it's optional. I say "in theory", because it's almost certainly something that the library caller will want to add to the context to get more information for cases where things are breaking, but that's still up to the caller.

It's designed to support the case where the caller didn't provide an artifact writer gracefully. Our CLI execution enforces the artifact Writer must be in the context before it hits this.

Reply:
That makes sense that in the CLI package we have an error if `artifactsWriter` is nil, this is the code I was looking for, thanks for pointing to it!.

Reply:
@bcrochet for some reason I have to reply to your 7LoC question in this thread. 

In theory, yep, as long as we want to do a "WriteIfNotNil". I imagined this would be a library. I should have included it here, but didn't. I could do that here if we feel strongly about it, or in a future PR if we're open to that.